pkg/controller/account/account_controller.go

```diff
@@ -605,6 +604,27 @@ func CreateAccount(reqLogger logr.Logger, client awsclient.Client, accountName,
 	return accountStatus, nil
 }
 
+func checkAWSAccountsLimitReached(r *ReconcileAccount, reqLogger logr.Logger, currentAccounts int) (bool, error) {
```

Comment:
Can we add some unit tests with this function before we merge it?

Reply:
Let me know if you want to :pear: on the testing if you're getting blocked or something.

Reply:
Good point! I'll add some in my next PR.

Reply:
I will take care of this request on a separate PR https://issues.redhat.com/browse/OSD-4073